### PR TITLE
[KYUUBI #1032] Keep server session id and engien session id consistent

### PR DESCRIPTION
**Pull request by ulysses-you**
### _Why are the changes needed?_

Closes https://github.com/apache/incubator-kyuubi/issues/1032

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala [hunk 1]
```diff
@@ -111,7 +110,7 @@ private[kyuubi] class EngineRef private(conf: KyuubiConf, user: String, sessionI
    */
   @VisibleForTesting
   private[kyuubi] lazy val engineSpace: String = shareLevel match {
-    case CONNECTION => ZKPaths.makePath(s"${serverSpace}_$shareLevel", appUser, sessionId)
+    case CONNECTION => ZKPaths.makePath(s"${serverSpace}_$shareLevel", appUser, namespaceId)
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala:113]: update the comment

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala [hunk 2]
```diff
@@ -93,8 +106,10 @@ class KyuubiSessionImpl(
 
   override def close(): Unit = {
     super.close()
-    sessionManager.operationManager.removeConnection(handle)
-    sessionManager.credentialsManager.removeSessionCredentialsEpoch(handle.identifier.toString)
+    if (handle != null) {
+      sessionManager.operationManager.removeConnection(handle)
+      sessionManager.credentialsManager.removeSessionCredentialsEpoch(handle.identifier.toString)
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala:111]: can we use `handle` as keys? @zhouyifan279

**Comment by yaooqinn** [comment]
cc @timothy65535

File: kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala [hunk 3]
```diff
@@ -100,9 +100,9 @@ object HighAvailabilityConf {
     .checkValue(_ > 0, "Must be positive")
     .createWithDefault(Duration.ofSeconds(120).toMillis)
 
-  val HA_ZK_ENGINE_SESSION_ID: OptionalConfigEntry[String] =
-    buildConf("ha.engine.session.id")
-    .doc("The sessionId will be attached to zookeeper node when engine started, " +
+  val HA_ZK_ENGINE_REF_ID: OptionalConfigEntry[String] =
+    buildConf("ha.engine.ref.id")
+    .doc("The engine reference id will be attached to zookeeper node when engine started, " +
```
Comment (cfmcgrady) [inline kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/HighAvailabilityConf.scala:105]: nit: indent

**Review by cfmcgrady — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks for review, merging to master

File: kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala [hunk 4]
```diff
@@ -53,23 +53,27 @@ class KyuubiSessionImpl(
...
 
-  private val engine: EngineRef = EngineRef(sessionConf, user, handle)
+  private val engine: EngineRef = new EngineRef(sessionConf, user)
   private val sessionEvent = KyuubiSessionEvent(this)
   EventLoggingService.onEvent(sessionEvent)
 
   private var transport: TTransport = _
   private var client: KyuubiSyncThriftClient = _
 
+  private var _handle: SessionHandle = _
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala:63]:
```scala
  private final val _handle: SessionHandle = SessionHandle(protocol)
  override def handle: SessionHandle = _handle
```

This looks flaky and tricky, compared with `AbstractSession`, can you refine this?

**Pull request closed without merging**